File: 0038-count-and-say/0038-count-and-say.cpp
class Solution {
public:
    string countAndSay(int n) {
        string s = "1";
        if(n==1)
            return s;
        n--;
        while(n--){
            // cout<<"#"<<endl;
            int i = 0;
            int freq = 0;
            char pre;
            string t ="";
            while(i<s.length()){
                // cout<<"$"<<endl;
                if(i==0){
                    freq++;
                    pre = s[i];
                }
                else{
                    // cout<<"%"<<endl;
                    if(pre == s[i])
                        freq++;
                    else{
                        t.push_back(freq + '0');
                        t.push_back(pre);                        

                        pre = s[i];
                        freq = 1;
                    }
                }
                i++;
            }
            t.push_back(freq + '0');
            t.push_back(pre);
            // cout<<t<<endl;
            s = t;
        }
        return s;
    }
};

File: 0038-count-and-say/0038-count-and-say.py
class Solution:
    def countAndSay(self, n: int) -> str:
        s = "1"
        if (n == 0 ):
           return s

        n -= 1
        while n > 0:
            i =0 
            freq = 0
            pre = '0'
            t = ""
            while i<len(s):
                if i == 0:
                    pre = s[i]
                    freq += 1
                else:
                    if pre == s[i]:
                        freq += 1
                    else:
                        t += str(freq)
                        t += pre

                        pre = s[i]
                        freq = 1
                
                i += 1
            t += str(freq)
            t += pre
            s = t
            n -= 1

        return s